harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -40,47 +50,37 @@ class Checkpoint(object):
     def __init__(
         self,
         session: session.Session,
+        task_id: str,
+        allocation_id: str,
         uuid: str,
-        experiment_config: Dict[str, Any],
-        experiment_id: int,
-        trial_id: int,
-        hparams: Dict[str, Any],
-        batch_number: int,
-        start_time: str,
-        end_time: str,
+        report_time: Optional[str],
         resources: Dict[str, Any],
-        validation: Dict[str, Any],
         metadata: Dict[str, Any],
-        determined_version: Optional[str] = None,
-        framework: Optional[str] = None,
-        format: Optional[str] = None,  # noqa: A002
-        model_version: Optional[int] = None,
-        model_id: Optional[int] = None,
+        state: CheckpointState,
+        training: Optional[CheckpointTrainingMetadata] = None,
     ):
         self._session = session
+        self.task_id = task_id
+        self.allocation_id = allocation_id
         self.uuid = uuid
-        self.experiment_config = experiment_config
-        self.experiment_id = experiment_id
-        self.trial_id = trial_id
-        self.hparams = hparams
-        self.batch_number = batch_number
-        self.start_time = start_time
-        self.end_time = end_time
+        self.report_time = report_time
         self.resources = resources
-        self.validation = validation
-        self.framework = framework
-        self.format = format
-        self.determined_version = determined_version
-        self.model_version = model_version
-        self.model_id = model_id
         self.metadata = metadata
+        self.state = state
+        self.training = training
+
+    def get_training(self) -> CheckpointTrainingMetadata:
+        if self.training is None:
+            raise RuntimeError("Non-training checkpoints are not yet supported.")
+        return self.training
```

Comment:
1. I don't think such getters are necessary when there's no extra I/O (like network fetch) is involved. so maybe use `@property`? 
2. looking at `checkpoint.get_training()` or `checkpoint.training` it's not clear for me what the result is. so maybe `training_metadata`? i.e.:


```suggestion
    @property
    def training_metadata(self) -> CheckpointTrainingMetadata:
        if self._training is None:
            raise RuntimeError("Non-training checkpoints are not yet supported.")
        return self._training
```

3. how can a user tell from API if a checkpoint has this training metadata? do we need a `has_training_metadata` method? why is throwing an exception here necessary? Maybe no need for this property/getter at all, just have it `None` when it's not available?

Reply:
This is a good point. In CLI I am going to check for Noneness and render empty string when training is None. There is one usage, in experimental.Checkpoint.download() that relies on it being a training checkpoint to get to the checkpoint_storage. I am going to modify it; let me know what you think.